extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -2198,4 +2214,57 @@ Runnable updateCurrentAndLatestOffsets()
       }
     };
   }
+
+  private Map<String, Object> getCurrentTotalStats() throws InterruptedException, ExecutionException, TimeoutException
+  {
+    Map<String, Object> allStats = Maps.newHashMap();
+    final List<ListenableFuture<Void>> futures = new ArrayList<>();
+
+    for (int groupId : taskGroups.keySet()) {
+      Map<String, Object> groupStats = (Map<String, Object>) allStats.get(String.valueOf(groupId));
+      if (groupStats == null) {
+        groupStats = Maps.newHashMap();
+        allStats.put(String.valueOf(groupId), groupStats);
+      }
+
+      final Map<String, Object> eGroupStats = groupStats;
```

Comment:
this is concurrently modified from who knows how many threads. Can the `Futures.allAsList` result be used instead? something like a `Stream` of `Pair<String, Map<String, Object>>` with a `Collectors.toMap` should work

Reply:
changed this to return a result object from each future, map is built only after all results are retrieved